File: src/api/MujDomecek.Domain/Aggregates/Zaznam/Media.cs
using MujDomecek.Domain.Abstractions;
using MujDomecek.Domain.ValueObjects;

namespace MujDomecek.Domain.Aggregates.Zaznam;

public sealed class Media : SoftDeletableEntity<Guid>
{
    public OwnerType OwnerType { get; set; }

    public Guid OwnerId { get; set; }

    public MediaType Type { get; set; }

    public string StorageKey { get; set; } = string.Empty;

    public string? OriginalFileName { get; set; }

    public string MimeType { get; set; } = string.Empty;

    public long SizeBytes { get; set; }

    public string? Description { get; set; }
}

File: src/web/src/lib/api/media.ts
import { api } from './client';
import type { MediaDto, MediaOwnerType, MediaType } from './types';

export interface MediaListResponse {
  items: MediaDto[];
}

export interface AddMediaRequest {
  ownerType: MediaOwnerType;
  ownerId: string;
  storageKey: string;
  mediaType: MediaType;
  originalFileName?: string;
  mimeType: string;
  sizeBytes: number;
  caption?: string;
}

export interface UpdateMediaRequest {
  caption?: string;
}

export interface MediaUrlResponse {
  url: string;
  expiresAt: string;
}

export const mediaApi = {
  async list(ownerType: MediaOwnerType, ownerId: string): Promise<MediaListResponse> {
    const params = new URLSearchParams({ ownerType, ownerId });
    return api.get<MediaListResponse>(`/media?${params.toString()}`);
  },

  async create(data: AddMediaRequest): Promise<MediaDto> {
    return api.post<MediaDto>('/media', data);
  },

  async update(id: string, data: UpdateMediaRequest): Promise<MediaDto> {
    return api.put<MediaDto>(`/media/${id}`, data);
  },

  async delete(id: string): Promise<void> {
    return api.delete(`/media/${id}`);
  },

  async getUrl(id: string): Promise<MediaUrlResponse> {
    return api.get<MediaUrlResponse>(`/media/${id}/url`);
  }
};
